File: Sorting/Cpp/Shorting/Selection_Sort.cpp
#include<iostream>
using namespace std;

// main function
int main(){

    // declaring variables
    int arr[20], pos, i, j, temp, num;

    // taking input of number-of-elemets
    cout << "Enter number of elements: ";
    cin >> num;

    // taking input of elements
    cout << endl << "Enter elements:" << endl;
    for(i = 0; i < num; i++)
        cin >> arr[i];

    // sorting elements using Selection Sort
    for (i = 0; i < num; i++)  {  
        pos = i;  
        for (j = i + 1; j < num; j++)  
        if (arr[j] < arr[pos])  
            pos = j;  
  
        temp = arr[pos];
        arr[pos] = arr[i];
        arr[i] = temp;  
    }  

    // printing Sorted elements
	cout << endl << "Sorted elements: " << endl;
    for(i = 0; i < num; i++)
        cout << arr[i] << endl;
    
    // returning 0
    return 0;
}
// main closed

File: Sorting/Java/selectionSort.java
class Sorting{
    public static void sort(int arr[]){
        int n = arr.length;
        for(int i = 0; i < n-1; i++){
            int min_idx = i;
            for(int j = i+1; j<n; j++){
                if(arr[j] < arr[min_idx] ){
                    min_idx = j;
                }
            }
            int temp = arr[min_idx];
            arr[min_idx] = arr[i];
            arr[i] = temp;

        }
    }
    public static void printArray(int arr[]){
        int n = arr.length;
        for(int i = 0; i < n; i++){
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }
    public static void main(String args[])  {
        int arr[] = {6,1,3,5,2,4};
        System.out.println("Array before sorting: ");
        printArray(arr);
        sort(arr); 
        System.out.println("\nArray after sorting: ");
        printArray(arr);
    }
}